JitsiParticipant.js

```diff
@@ -61,6 +62,32 @@ export default class JitsiParticipant {
             true);
     }
 
+    /**
+     * Sets source info.
+     * @param {MediaType} mediaType The media type, 'audio' or 'video'.
+     * @param {boolean} muted The new muted state.
+     * @param {string} sourceName The name of the source.
+     * @param {string} videoType The video type of the source.
+     * @returns {void}
+     */
+    _setSources(mediaType, muted, sourceName, videoType) {
+        let sourceByMediaType = this._sources.get(mediaType);
+        const sourceInfo = {
+            muted,
+            videoType
+        };
+
+        if (sourceByMediaType?.size) {
+            sourceByMediaType.set(sourceName, sourceInfo);
+
+            return;
+        }
+
+        sourceByMediaType = new Map();
+        sourceByMediaType.set(sourceName, sourceInfo);
+        this._sources.set(mediaType, sourceByMediaType);
```

Comment:
Am I reading this correctly, that we'll have this Map with media type (so audio, video) pointing to another map, with the source infos, indexes by source name?

Wouldn't this make it simpler?

```
this._sources = {
    audio: new Map(),
    video: new Map()
};
```

Since the keys for the outermost map are always constant, are they not?

Reply:
Yes, it does make it simpler but then also creates the need for declaring a new type for this Object here (when we transition to TS) and in [JM](https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/participants/types.ts#L54).  The current approach doesn't sound too complicated for me, we do it the same way for remote tracks in TPC but do let me know if you have a strong preference.

Reply:
No strong preference, and since it's an internal thing it can be changed anytime if we have a change of heart.